### PR TITLE
GitHub sync

### PR DESCRIPTION
**Pull request by nshahan**
Updated gallery branch with new Material Stepper examples:
https://github.com/dart-lang/angular_components_example/tree/github-sync

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,26 @@
...
+ * Material Checkbox: Migrate from the deprecated Glyph component to Material
+   Icon.
+ * Material Input: Don't override unfocused underline color when the input is
+   invalid.
+ * Material Select:
+   * Support focus on dropdown button when dismissed via keyboard.
+   * Fix focus bug when clicking on a different focusable element.
+ * Add new CSS class names used by `ShowHideDirective` to avoid conflicts with
+   Angular JS apps.
+ * Only focus when `_autoFocus` is `true` when using `AutoFocusDirective`.
```
Comment (kevmoo) [inline CHANGELOG.md]: `_autoFocus` is private. Is there a public flag/option/something here?

Comment (TedSander) [reply CHANGELOG.md]: There is a public setter for autoFocus which is also an input. Probably just remove the _ in the release notes.

File: lib/utils/angular/scroll_host/angular_2.dart [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+  void ngOnDestroy() {
+    _scrollHost.stickyController.untrackFloating(_element);
+  }
+}
+
+/// A directive to dynamically set a class based on the pan state.
+///
+/// Example use:
+///
+///     <some-element acxPanClass="'pan-style'" ...
```
Comment (chalin) [inline lib/utils/angular/scroll_host/angular_2.dart]: [PREFER backtick fences for code blocks.](https://www.dartlang.org/guides/language/effective-dart/documentation#prefer-backtick-fences-for-code-blocks)

File: lib/utils/angular/scroll_host/angular_2.dart [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+/// A directive to sticky an Element.
+///
+/// Example use:
+///
+///     <div [acxStickyElement]="endsticky">
+///       Your awesome content
+///     </div>
+///     Other awesome content
+///     <div #endsticky></div> <!-- Where sticky should end -->
```
Comment (chalin) [inline lib/utils/angular/scroll_host/angular_2.dart]: [PREFER backtick fences for code blocks.](https://www.dartlang.org/guides/language/effective-dart/documentation#prefer-backtick-fences-for-code-blocks)

File: lib/utils/angular/scroll_host/angular_2.dart [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+      DomService domService, ScrollHost scrollHost, ElementRef elementRef)
+      : super(domService, scrollHost, elementRef.nativeElement);
+
+  @override
+  void ngOnInit() => startPanListener();
+
+  @override
+  void ngOnDestroy() => stopPanListener();
+
+  // Overridden to add Input annotation and keep angular2 dep outside of base
```
Comment (chalin) [inline lib/utils/angular/scroll_host/angular_2.dart]: angular2 --> Angular

(Should this file be renamed as well, given that it is `angular_2.dart`?)

File: lib/utils/angular/scroll_host/angular_2.dart [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+  // class.
+  @override
+  @Input('acxPanClass')
+  set className(String value) => super.className = value;
+}
+
+/// A directive to sticky an Element.
```
Comment (chalin) [inline lib/utils/angular/scroll_host/angular_2.dart]: I can't make sense of this doc comment. Maybe:

> A directive to stick an Element.

File: lib/utils/angular/scroll_host/angular_2.dart [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+///     <div [acxStickyElement]="endsticky">
+///       Your awesome content
+///     </div>
+///     Other awesome content
+///     <div #endsticky></div> <!-- Where sticky should end -->
+///
+/// Note that adding the acxStickyElement to an element that has ngIf has been
+/// known to not work correctly. Suggestion is to have the ngIf and
+/// acxStickyElements on different html elements (i.e., add an extra outer div
+/// for the ngIf).
```
Comment (chalin) [inline lib/utils/angular/scroll_host/angular_2.dart]: Code elements should be marked as code in the lines above. E.g.,

```dart
/// Note that adding the `acxStickyElement` to an element that has `ngIf` has been ...
```

File: lib/utils/angular/scroll_host/angular_2.dart [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+
+  StickyElementDirective(this._scrollHost, this._stickyElement);
+
+  /// The end element that determines the range of the sticky element.
+  @Input('acxStickyElement')
+  set endElement(Element element) {
+    _endElement = element;
+  }
+
+  /// Optional css class to use when element is stuck.
```
Comment (chalin) [inline lib/utils/angular/scroll_host/angular_2.dart]: nit: css -> CSS

File: lib/material_stepper/common.dart [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+const String HORIZONTAL = 'horizontal';
+const String VERTICAL = 'vertical';
+const String SIZE_DEFAULT = 'default';
+const String SIZE_MINI = 'mini';
+const String ALL = 'all';
+const String BACKWARDS = 'backwards';
+const String NONE = 'none';
```
[the remark this answers is not in the document]
Comment (nshahan) [reply lib/material_stepper/common.dart]: Cleaning up the constant names will take a little while. I don't want to block this PR on that change but I will look into making them more consistent.

**Comment by nshahan** [comment]
I'll add the comment nits to our documentation fixit we have coming up.

File: lib/material_stepper/common.dart (hunk 8 above)
[the remark this answers is not in the document]
Comment (chalin) [reply lib/material_stepper/common.dart]: > Cleaning up the constant names will take a little while. I don't want to block this PR on that change but I will look into making them more consistent.

👍

**Pull request merged**